api/src/services/listing.service.ts

```diff
@@ -965,6 +967,68 @@ export class ListingService implements OnModuleInit {
     return mapTo(Listing, rawListing);
   }
 
+  async duplicate(
+    dto: ListingDuplicate,
+    requestingUser: User,
+  ): Promise<Listing> {
+    const storedListing = await this.findOrThrow(
+      dto.storedListing.id,
+      ListingViews.details,
+    );
+
+    await this.permissionService.canOrThrow(
+      requestingUser,
+      'listing',
+      permissionActions.create,
+      {
+        jurisdictionId: storedListing.jurisdictions.id,
+      },
+    );
+
+    if (dto.name.trim() === storedListing.name) {
+      throw new BadRequestException('New listing name must be unique');
+    }
+
+    const mappedListing = mapTo(ListingCreate, storedListing);
+
+    const listingEvents = mappedListing.listingEvents?.filter(
+      (event) => event.type !== ListingEventsTypeEnum.lotteryResults,
+    );
+
+    const listingImages = mappedListing.listingImages?.map((unsavedImage) => {
+      return {
+        assets: {
+          fileId: unsavedImage.assets.fileId,
+          label: unsavedImage.assets.label,
+        },
+        ordinal: unsavedImage.ordinal,
+      };
+    });
+
+    if (!dto.includeUnits) {
+      delete mappedListing['units'];
+    }
+
+    const newListingData: ListingCreate = {
+      ...mappedListing,
+      name: dto.name,
+      status: ListingsStatusEnum.pending,
+      listingEvents: listingEvents,
+      listingMultiselectQuestions:
+        storedListing.listingMultiselectQuestions?.map((question) => {
+          return {
+            id: question.multiselectQuestionId,
+            ordinal: question.ordinal,
+          };
+        }),
+      listingImages: listingImages,
+      lotteryLastRunAt: undefined,
```

Comment:
Just merged in a PR adding lotteryLastPublishedAt field which should be set to undefined here too